src/librustdoc/clean/types.rs

```diff
@@ -1652,7 +1649,7 @@ impl PrimitiveType {
             sym::pointer => Some(PrimitiveType::RawPointer),
             sym::reference => Some(PrimitiveType::Reference),
             kw::Fn => Some(PrimitiveType::Fn),
-            sym::never => Some(PrimitiveType::Never),
+            sym::never | sym::never_type => Some(PrimitiveType::Never),
```

Comment:
`sym::never_type == "never_type"`, so this isn't right. I think you may need to add a `sym::bang == "!"` symbol.

Reply:
Good point!

Reply:
Fixed.